src/idl_gen_swift.cpp

```diff
@@ -0,0 +1,719 @@
+
+    code_.SetValue("TYPE", GenType(key_field.value.type));
+    code_ +=
+        "\tpublic static func lookupByKey(vector: Int32, key: {{TYPE}}, fbb: "
+        "ByteBuffer) -> {{VALUENAME}}? {";
+
+    if (key_field.value.type.base_type == BASE_TYPE_STRING)
+      code_ += spacing + "let key = key.utf8.map { $0 }";
+    code_ += spacing +
+             "var span = fbb.read(def: Int32.self, position: Int(vector - 4))";
+    code_ += spacing + "var start: Int32 = 0";
+    code_ += spacing + "while span != 0 {";
+    code_ += double_spacing + "var middle = span / 2";
+    code_ +=
+        double_spacing +
+        "let tableOffset = Table.indirect(vector + 4 * (start + middle), fbb)";
+    if (key_field.value.type.base_type == BASE_TYPE_STRING) {
+      code_ += double_spacing + "let comp = Table.compare(" + offset_reader +
+               ", key, fbb: fbb)";
+    } else {
+      code_ += double_spacing +
+               "let comp = fbb.read(def: {{TYPE}}.self, position: Int(" +
+               offset_reader + "))";
+    }
+
+    code_ += double_spacing + "if comp > 0 {";
+    code_ += double_spacing + "\tspan = middle";
+    code_ += double_spacing + "} else if comp < 0 {";
+    code_ += double_spacing + "\tmiddle += 1";
+    code_ += double_spacing + "\tstart += middle";
+    code_ += double_spacing + "\tspan -= middle";
+    code_ += double_spacing + "} else {";
+    code_ += double_spacing + "\treturn {{VALUENAME}}(fbb, o: tableOffset)";
+    code_ += double_spacing + "}";
+    code_ += spacing + "}";
+    code_ += spacing + "return nil";
+    code_ += "\t}";
+  }
+
+  std::string GenOffset() { return "let o = {{ACCESS}}.offset({{OFFSET}}); "; }
+
+  std::string GenReaderMainBody(std::string optional = "") {
+    return "\tpublic var {{VALUENAME}}: {{VALUETYPE}}" + optional + " { ";
+  }
+
+  std::string GenReader(std::string type, std::string at = "{{OFFSET}}") {
+    return "{{ACCESS}}.readBuffer(of: {{" + type + "}}.self, at: " + at + ")";
+  }
+
+  std::string GenConstructor(
+      std::string offset = "{{ACCESS}}.postion + {{OFFSET}}") {
+    return "{{VALUETYPE}}({{ACCESS}}.bb, o: " + offset + ") }";
+  }
+
+  std::string GenMutate(std::string offset, std::string get_offset = "",
+                        bool isRaw = false) {
+    return "\tpublic func mutate({{VALUENAME}}: {{VALUETYPE}}) -> Bool {" +
+           get_offset + " return {{ACCESS}}.mutate({{VALUENAME}}" +
+           (isRaw ? ".rawValue" : "") + ", index: " + offset + ") }";
+  }
+
+  std::string GenMutateArray() {
+    return "\tpublic func mutate({{VALUENAME}}: {{VALUETYPE}}, at index: "
+           "Int32) -> Bool { " +
+           GenOffset() +
+           "return {{ACCESS}}.directMutate({{VALUENAME}}, index: "
+           "{{ACCESS}}.vector(at: o) + index * {{SIZE}}) }";
+  }
+
+  std::string GenEnumBaseContructor() {
+    return "{{VALUETYPE}}(rawValue: {{CONSTANT}})!";
+  }
+
+  std::string GenEnumConstructor(std::string at = "{{OFFSET}}") {
```

Comment:
V813 Decreased performance. The 'at' argument should probably be rendered as a constant reference. idl_gen_swift.cpp 592

It called with the default only once.